lib/asciidoctor/js/asciidoctor_ext/browser/abstract_node.rb

```diff
@@ -0,0 +1,61 @@
+module Asciidoctor
+class AbstractNode
+  def generate_data_uri_from_uri image_uri, cache_uri = false
+    %x{
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', #{image_uri}, false)
+      // the response type cannot be changed for synchronous requests made from a document
+      // xhr.responseType = 'arraybuffer'
+      xhr.overrideMimeType('text/plain; charset=x-user-defined')
+
+      var contentType = ''
+      var b64encoded = ''
+      var status = -1
+
+      try {
+        xhr.addEventListener('load', function() {
+          status = this.status
+          // status is 0 for local file mode (i.e., file://)
+          if (status === 0 || status === 200) {
+            var binary = ''
+            var rawText = this.responseText;
+            for (var i = 0, len = rawText.length; i < len; ++i) {
+              var c = rawText.charCodeAt(i)
+              var byteCode = c & 0xff // byte at offset i
+              binary += String.fromCharCode(byteCode)
+            }
+            b64encoded = btoa(binary)
+            contentType = this.getResponseHeader('content-type')
+          }
+        })
+        xhr.send(null)
+        // try to detect the MIME Type from the file extension
+        if (!contentType) {
```

Comment:
`content-type` is undefined when using the `file` protocol.
The above code will try to detect the MIME type from the file extension. It can return an incorrect value on edge cases but overall I think it provides a better user experience.